src/utils/cspUtils.js

```diff
@@ -0,0 +1,133 @@
+// Supported CSP checks are img-src, frame-src, media-src, object-src script-src-elem 
+// Supported fallbacks are child-src, script-src, and default-src
+
+import toml from 'toml';
+import axios from 'axios';
+import Policy from 'csp-parse';
+import _ from 'lodash';
+import { isLinkInternal } from '../utils';
+import cheerio from 'cheerio';
+import escapeStringRegexp from 'escape-string-regexp';
+
+/* Helper function to retrieve the netlify.toml from repo */
+async function _parseNetlifyToml(repoName) {
+  // axios get withCredentials false is required https://stackoverflow.com/questions/34078676/access-control-allow-origin-not-allowed-when-credentials-flag-is-true-but/34099399 
+  const tomlUrl = `https://raw.githubusercontent.com/isomerpages/${repoName}/staging/netlify.toml`;
+  const tomlFile = await axios.get(tomlUrl, { withCredentials: false } ); 
+  return toml.parse(tomlFile.data);
+};
+
+export async function getCSP(repoName) {
+  const tomlData = await _parseNetlifyToml(repoName);
+  const csp = tomlData.headers[0].values['Content-Security-Policy'];
+  return csp;
+};
+
+function _stringContainsValue(string, value) {
+  const VALUE_REGEX = `(\\s|^)${value}(\\s|$)`;
+  const ValueRegexTest = new RegExp(VALUE_REGEX);
+  return ValueRegexTest.test(string);
+};
```

Comment:
The regex checks specifically for if these values are preceded by whitespace or are at the start/ end of the line, for example in the case of the scheme-source `https:`, the regex returns true for `https:` and false for `https://www.google.com`. 

Correct me if I'm wrong but this does not seem possible with `includes`, or it seems more complicated with `includes`?

Reply:
Ah ok I see what you mean. My only (slight) concern is that the regex isn't as easy to read, but I think it's okay as it is.
An alternative would be to return the policy as an array (split on a whitespace ` `) so that this function would be `_policyContainsValue` and you can simply search `arr.includes(value)`.

Reply:
77011dc